### PR TITLE
JSON API for wiki pages #1069

### PR DESCRIPTION
**Pull request by 500swapnil**
Make sure these boxes are checked before your pull request is ready to be reviewed and merged. Thanks!

* [x] all tests pass -- `rake test:all`
* [ ] code is in uniquely-named feature branch, and has been rebased on top of latest master (especially if you've been asked to make additional changes)
* [ ] pull request is descriptively named with #number reference back to original issue
* [ ] if possible, multiple commits squashed if they're smaller changes
* [ ] reviewed/confirmed/tested by another contributor or maintainer
* [ ] `schema.rb.example` has been updated if any database migrations were added

Please be sure you've reviewed our contribution guidelines at https://publiclab.org/wiki/contributing-to-public-lab-software

We have a loose schedule of reviewing and pulling in changes every Tuesday and Friday, and publishing changes on Fridays. Please alert developers on plots-dev@googlegroups.com when your request is ready or if you need assistance.

Thanks!

### CODE REVIEW

File: app/controllers/tag_controller.rb [hunk 1]
```diff
@@ -41,13 +41,18 @@ def show
                         .page(params[:page])
                         .order("node_revisions.timestamp DESC")
     end
-
+    respond_with(@nodes) do |format|
```
Comment (jywarren) [inline app/controllers/tag_controller.rb]: This looks good, but I think this has to go down on line 51, at the end of the action method, or some of the variables used in the templates won't exist when we try to render it. Make sense? Thanks!

Comment (500swapnil) [inline app/controllers/tag_controller.rb]: Okay will do it.

File: app/controllers/tag_controller.rb [hunk 2]
```diff
@@ -41,13 +41,18 @@ def show
...
     end
-
     @notes = nodes.where('node.nid NOT IN (?)', qids) if @node_type == "note"
     @questions = nodes.where('node.nid IN (?)', qids) if @node_type == "questions"
     @wikis = nodes if @node_type == "wiki"
     @nodes = nodes if @node_type == "maps"
     @title = params[:id]
     set_sidebar :tags, [params[:id]]
+    
+    respond_with(@nodes) do |format|
```
Comment (jywarren) [inline app/controllers/tag_controller.rb]: Oops, and i mistyped -- this should be `@notes`, not `@nodes`. Sorry! and on lines 53-4.

Comment (jywarren) [inline app/controllers/tag_controller.rb]: apologies!

**Comment by 500swapnil** [comment]
Is it ok now?

**Comment by jywarren** [comment]
It looks good, lets see how the tests do! Thanks!

**Comment by jywarren** [comment]
Hmm, ok your test didn't pass:

```ruby
================================================================================
Error: test_tag_show_JSON(TagControllerTest): TypeError: no implicit conversion of ActionController::TestResponse into String
test/functional/tag_controller_test.rb:110:in `block in <class:TagControllerTest>'
```

Let's see...

**Comment by 500swapnil** [comment]
What should I do?

**Comment by jywarren** [comment]
Oh, you know what, i think maybe it needs to be:

```ruby
json = ActiveSupport::JSON.decode(@response.body)
```

Can you try that?

**Comment by jywarren** [comment]
I'll be in transit for 15m, but can check on this later, thanks!

**Comment by jywarren** [comment]
Perfect; merging now! 🎉  

If you're looking for a new challenge, please take a look at our `help-wanted` list: https://github.com/publiclab/plots2/labels/help-wanted

Thanks!

Oh, and also -- for future reference, it's important to do your development on a named feature branch -- you can read more here:

https://publiclab.org/wiki/contributing-to-public-lab-software#A+sample+git+workflow

And there are instructions on rewinding back your master branch here:

https://publiclab.org/wiki/contributing-to-public-lab-software#Rewinding+the+master+branch

Thanks again!!!

**Comment by 500swapnil** [comment]
Okay thank you!!

**Comment by jywarren** [comment]
I think that may be record time for a new issue being filed and fixed --
thank YOU!

**Pull request merged**